### PR TITLE
update the tensor.scatter_ doc

### LINKED ISSUE #119543: `Tensor.scatter_` unclear documentation
### 📚 The doc issue

The following code works:

```python
import torch

x = torch.tensor([1, 2])
index=torch.tensor([0, 1])

x.scatter_(dim=0, index=index, value=0)
print(x)  # tensor([0, 0])
```
`torch.Tensor.scatter_` allows the use of a `value` keyword arg which is undocumented https://pytorch.org/docs/stable/generated/torch.Tensor.scatter_.html. 

It seems this is the mechanism through which `src` is able to allow floats, as well as tensors. But this would imply someone could do something like 
```python
x.scatter_(dim=dim, index=index, src=0.0)
```
Though this will fail, with an error saying `src` needs to be a tensor, contradicting the docs for this method.

### Suggest a potential alternative/fix

The docs could be a bit clearer, mainly in reference to the parameter type hint `src (Tensor or float)`, which isn't entirely true.

cc @svekars @brycebortree @mikaylagawarecki

### PR DESCRIPTION
**Pull request by lancerts**
Fixes #119543 

- doc fixed with the `reduce` being a kwarg (see below for details)
- doc added another interface `(int dim, Tensor index, Number value, *, str reduce)` where
the full signature in the pyi file after build is
```
def scatter_(self, dim: _int, index: Tensor, value: Union[Number, _complex], *, reduce: str) -> Tensor: 
```
. This can be further verified in
https://github.com/pytorch/pytorch/blob/02fb04352275d18529fd10741a0626f0bc3eab5e/aten/src/ATen/native/native_functions.yaml#L8014

Therefore, the value can be int, bool, float, or complex type.

Besides the issue mentioned in 119543, the `reduce should be a kwarg` as shown below
```
 * (int dim, Tensor index, Tensor src)
 * (int dim, Tensor index, Tensor src, *, str reduce)
 * (int dim, Tensor index, Number value)
 * (int dim, Tensor index, Number value, *, str reduce)
 ```

The test case for scala value is already implemented in

https://github.com/pytorch/pytorch/blob/70bc3b3be4d8aad3029e7270ec37e23f95b3c39f/test/test_scatter_gather_ops.py#L86

so no additional test case required.

@mikaylagawarecki  @janeyx99 

cc @albanD

### CODE REVIEW

**Comment by lancerts** [comment]
@mikaylagawarecki lmk if things look ok, thanks

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -4473,6 +4475,31 @@ def callable(a, b) -> number
...
+Writes the value from :attr:`value` into :attr:`self` at the indices
+specified in the :attr:`index` tensor.  This operation is equivalent to the previous version,
+with the :attr:`src` tensor filled entirely with :attr:`value`.
+
+Args:
+    dim (int): the axis along which to index
+    index (LongTensor): the indices of elements to scatter, can be either empty
+        or of the same dimensionality as ``src``. When empty, the operation
+        returns ``self`` unchanged.
+    value (int, float, bool or complex): the value to scatter.
```
Comment (mikaylagawarecki) [inline torch/_tensor_docs.py:4490]:
```suggestion
    value (Scalar): the value to scatter.
```

File: torch/_tensor_docs.py [hunk 2]
```diff
@@ -4473,6 +4475,31 @@ def callable(a, b) -> number
     tensor([[2.0000, 2.0000, 3.2300, 2.0000],
             [2.0000, 2.0000, 2.0000, 3.2300]])
 
+.. function:: scatter_(dim, index, value, *, reduce: str) -> Tensor:
```
Comment (mikaylagawarecki) [inline torch/_tensor_docs.py:4478]: nit:
```suggestion
.. function:: scatter_(dim, index, value, *, reduce=None) -> Tensor:
```

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -4473,6 +4475,31 @@ def callable(a, b) -> number
...
+    index (LongTensor): the indices of elements to scatter, can be either empty
+        or of the same dimensionality as ``src``. When empty, the operation
+        returns ``self`` unchanged.
+    value (int, float, bool or complex): the value to scatter.
+
+Keyword args:
+    reduce (str, optional): reduction operation to apply, can be either
+        ``'add'`` or ``'multiply'``.
+
+Example::
```
Comment (mikaylagawarecki) [inline torch/_tensor_docs.py:4496]: nit: I think you need a newline below Example:: so it will render with 1 colon

**Review by mikaylagawarecki — APPROVED** [review]
Thanks for the fix! Just some nits

File: torch/_tensor_docs.py (hunk 3 above)
Comment (lancerts) [reply torch/_tensor_docs.py:4496]: Added a newline

**Comment by lancerts** [comment]
> Thanks for the fix! Just some nits

@mikaylagawarecki nit all addressed, thanks for the suggestions.
